src/symbolize/gimli/parse_running_mmaps_unix.rs

```diff
@@ -141,13 +162,73 @@ impl FromStr for MapsEntry {
     }
 }
 
+#[cfg(target_os = "nto")]
+pub fn config() -> (&'static str, usize) {
+    ("/proc/self/pmap", 1)
+}
+
+#[cfg(target_os = "nto")]
+impl FromStr for MapsEntry {
+    type Err = &'static str;
+
+    // Format: vaddr,size,flags,prot,maxprot,dev,ino,offset,rsv,guardsize,refcnt,mapcnt,path
+    // e.g.: "0x00000022fa36b000,0x0000000000002000,0x00000071,0x05,0x0f,0x0000040b,0x00000000000000dd,
+    //        0x0000000000000000,0x0000000000000000,0x00000000,0x00000005,0x00000003,/proc/boot/cat"
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        let mut parts = s.split(',');
+        let vaddr_str = parts.next().ok_or("Couldn't find virtual address")?;
+        let size_str = parts.next().ok_or("Couldn't find size")?;
+        let _flags_str = parts.next().ok_or("Couldn't find flags")?;
+        let prot_str = parts.next().ok_or("Couldn't find protection")?;
+        let _maxprot_str = parts.next().ok_or("Couldn't find maximum protection")?;
+        let dev_str = parts.next().ok_or("Couldn't find device")?;
+        let ino_str = parts.next().ok_or("Couldn't find inode")?;
+        let offset_str = parts.next().ok_or("Couldn't find offset")?;
+        let _rsv_str = parts.next().ok_or("Couldn't find reserved pages")?;
+        let _guardsize_str = parts.next().ok_or("Couldn't find guard size")?;
+        let _refcnt_str = parts.next().ok_or("Couldn't find reference count")?;
+        let _mapcnt_str = parts.next().ok_or("Couldn't find mapped count")?;
+        let pathname_str = parts.next().unwrap_or(""); // pathname may be omitted.
+
+        let hex =
+            |s: &str| usize::from_str_radix(&s[2..], 16).map_err(|_| "Couldn't parse hex number");
+        let address = { (hex(vaddr_str)?, hex(vaddr_str)? + hex(size_str)?) };
+
+        // TODO: Probably a rust'ier way of doing this
```

Comment:
You could probably cook something up using [`array::from_fn`](https://doc.rust-lang.org/stable/std/array/fn.from_fn.html) but I'm not going to block this on that.